[IMP] agreement,agreement_legal: support activities

yajo opened the pull request:
- The `agreement` model now features mail and activities from its base module.
- The `agreement_legal` module doesn't need to add mail support now.

@Tecnativa TT22477

pedrobaeza reviewed (commented):
Wasn't the chatter added in views by `agreement_legal` module?

yajo commented:
Yes, but I checked and it's a completely different form view that doesn't inherit from the form view defined in `agreement`.

pedrobaeza commented:
@max3903 can you approve it?

Commit:
[IMP] agreement,agreement_legal: support activities

- The `agreement` model now features mail and activities from its base module.
- The `agreement_legal` module doesn't need to add mail support now.
- Add activities to form views.

@Tecnativa TT22477

dreispt reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge minor

dreispt commented:
@OCA-git-bot You're too late. Pedro's minion bots were faster than you.

> This PR has the approved label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer).

pedrobaeza commented:
/ocabot merge minor
